vars/sconsBuild.groovy

```diff
@@ -1,50 +1,201 @@
+   *  If false, a failure of the scons commands will cause this step to fail.
+   */
 
-    githubNotify credentialsId: 'daos-jenkins-commit-status',
-                 description: env.STAGE_NAME,
-                 context: "build" + "/" + env.STAGE_NAME,
-                 status: "PENDING"
+    /* If we have to tamper with the checkout, we also need to remove
+     * the potential tampering before the scm operation.
+     */
+    if (config['scons_local_replace'] && config['target']) {
+       sh "rm -rf \"\${WORKSPACE}/${config['target']}/scons_local\""
+    }
+
+    scm_config = [withSubmodules: true]
+    if (config['scm']) {
+        scm_config = config['scm']
+        if (config['target'] && !scm_config['checkoutDir']) {
+            scm_config['checkoutDir'] = config['target']
+        }
+    }
+    checkoutScm(scm_config)
+    if (env.DAOS_JENKINS_NOTIFY_STATUS != null) {
+        githubNotify credentialsId: 'daos-jenkins-commit-status',
+                     description: env.STAGE_NAME,
+                     context: "build" + "/" + env.STAGE_NAME,
+                     status: "PENDING"
+    }
+
+    if (config['scons_local_replace'] && config['target']) {
+       // replace the scons_local directory in config['target']
+       // with a link to the scons_local directory.
+       sh """rm -rf "\${WORKSPACE}/${config['target']}/scons_local"
+             ln -s "${WORKSPACE}/scons_local" \
+               "\${WORKSPACE}/${config['target']}/scons_local"
+             """
+    }
+
+    set_cwd = ''
+    if (config['directory']) {
+        set_cwd = "cd ${config['directory']}\n"
+    } else if (config['target']) {
+        set_cwd = "cd ${config['target']}\n"
+    }
+
+    scons_args = ''
+    sconstruct = ''
+    if (config['sconstruct']) {
+        sconstruct = " --sconstruct=${config['sconstruct']}"
+        scons_args += sconstruct
+    }
+    if (config['update_prereq']) {
+        scons_args += " --update-prereq=${config['update_prereq']}"
+    } else {
+        scons_args += ' --update-prereq=all'
+    }
+    if (config['build_deps']) {
+        scons_args += " --build-deps=${config['build_deps']}"
+    } else {
+        scons_args += ' --build-deps=yes'
+    }
+    if (!config['no_install']) {
+        scons_args += ' install'
+    }
+    if (config['USE_INSTALLED']) {
+        scons_args += " USE_INSTALLED=${config['USE_INSTALLED']}"
+    } else {
+        scons_args += ' USE_INSTALLED=all'
+    }
+    if (config['SRC_PREFIX']) {
+        scons_args += " SRC_PREFIX=${config['SRC_PREFIX']}"
+    } else if (config['target']) {
+        scons_args += " SRC_PREFIX=\${WORKSPACE}}"
+    }
+    if (config['REQUIRES']) {
+        scons_args += " REQUIRES=${config['REQUIRES']}"
+    } else if (config['target']) {
+        scons_args += " REQUIRES=${config['target']}"
+    }
+    //scons -c is not perfect so get out the big hammer
+    clean_files = "_build.external* install build"
```

Comment:
Your earlier comment made me wonder if we should be this aggressive (i.e. dangerous) with an `rm -rf` or if we should be more targeted:
```suggestion
    clean_files = "_build.external{,-Linux} install build"
```

particularly considering this is going to be `clean_files = "_build.external install build"` very soon, once the _arch_ stuff is removed from iof and cart.

I'm on the fence but do like to be conservative and targeting when using destructive commands like `rm -rf`.

Reply:
Change made for next update.